src/pymachine/definition_parser.py

```diff
@@ -487,84 +472,78 @@ def __parse_expr(self, expr, root, loop_to_defendum=True,
         logging.debug(expr)
         raise pe
 
-    def __parse_definition(self, definition, root, loop_to_defendum=True,
-                           three_parts=False):
+    def __parse_definition(self, definition, root, loop_to_defendum=True):
         logging.debug(str(definition))
         for d in definition:
-            yield self.__parse_expr(d, root, loop_to_defendum, three_parts)[0]
+            yield self.__parse_expr(d, root, loop_to_defendum)[0]
 
-    def parse_into_machines(self, string, printname_index=0, add_indices=False,
-                            loop_to_defendum=True, three_parts=False):
-        printname = string.split('\t')[printname_index]
-        try:
-            id_, urob, pos, def_, comment = string.split('\t')[4:]
-        except:
-            raise Exception(string.split('\t'))
+    def parse_into_machines(self, printname, id_, def_, add_indices=False,
+                            loop_to_defendum=True):
 
         machine = self.create_machine(printname.lower(), 1)
-        #TODO =AGT -> partition 1, =PAT -> partition 2, =TO -> ?
 
         if add_indices:
             machine.printname_ = machine.printname() + id_sep + id_
 
-        if def_ != '':
-            logging.debug(def_)
-            parsed = self.parse(def_)
-            logging.debug(parsed)
-            for parsed_expr in self.__parse_definition(
-                    parsed[0], machine, loop_to_defendum, three_parts):
-                machine.append(parsed_expr, 0)
+        #if def_ != '':
+        logging.debug(def_)
+        parsed = self.parse(def_)
+        logging.debug(parsed)
+        for parsed_expr in self.__parse_definition(
+                parsed[0], machine, loop_to_defendum):
+            machine.append(parsed_expr, 0)
 
         self.unify(machine)
         return machine
 
-def read(f, plur_filn, printname_index=0, add_indices=False,
-         loop_to_defendum=True, three_parts=False):
```

Comment:
@makrai 
This currently breaks 4lang because of this line:
https://github.com/kornai/4lang/blob/master/src/fourlang/lexicon.py#L32

If you change the name and/or parameters of a function, you must search through all code to find these before you can safely commit a change.